stratified vs. random cross-val.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Wed Jul 14 18:29:08 2021
+
+@author: matthewayala
+"""
+
+import numpy as np
+from sklearn.linear_model import LogisticRegression
+import matplotlib.pyplot as plt
+from sklearn.model_selection import StratifiedKFold
+from sklearn.model_selection import KFold
+
+# Simulate logistic data with/without noise
+
+def logistic(x, b, noise=None):
+    L = x.T.dot(b)
+    if noise is not None:
+        L = L+noise
+    return 1/(1+np.exp(-L))
+
+# Function that sumulates logistic data, trains and validates model on the data.
+
+def get_noise_accuracy(noise_iterable,x_values,cut_off):
+    """Returns the noise iterable, the list of accuracy scores, and the list of standard deviations for error bars."""
+    bias = np.ones(len(x_values))
+    X = np.vstack([x_values,bias]) # Add Intercept
+    B =  [1., 1.] # Sigmoid params for X   
+    
+    mean_acc_scores = []
+    standard_deviations = []
+    noise_list = noise_iterable # Different noise values to be used in gaussian distribution sampling.
+    for i in noise_list:
+        # Creating simulated logistic data set with several noise values
+        pnoisy = logistic(X, B, noise=np.random.normal(loc=0., scale= i, size=len(x_values)))
+        
+        # Reshaping data for model    
+        x_data = x_values.reshape(len(x_values),1)
+        pnoisy_data = pnoisy.reshape(len(x_values),1)
+        transformed_p = [] # Empty list to hold new transformed data (1's and 0's)
+        for a in pnoisy_data:    
+            if a >= cut_off: # Turning values > 0.5 in 1's and < 0.5 into 0's
+                a = 1
+            else:
+                a = 0
+            transformed_p.append(a) # Adding transformed values to the empty list.
+        
+        # Turning labels into an array for stratifiedkfold
+        transformed_p = np.array(transformed_p)
+        # Creating Logistic regression instance
+        LogReg = LogisticRegression(max_iter=1000)
+        
+        # StratifiedKFold Splits
+        skf = StratifiedKFold(n_splits = 5)
+        # kf = KFold(n_splits = 5)
+        accuracy_scores = [] # Empty accuracy score list to be filled with each iteration of kfold
+        for train_index, test_index in skf.split(x_data,transformed_p):
```

Comment:
Instead of commenting out the call to `KFold` here, I'd suggest passing a bool argument to the function (call it `use_stratified_cv`), then choosing which split function to use here based on that argument:

```suggestion
        if use_stratified_cv:
            skf = StratifiedKFold(n_splits=5)
            split = skf.split(x_data, transformed_p)
        else:
            kf = KFold(n_splits=5)
            split = kf.split(x_data)
        accuracy_scores = [] # Empty accuracy score list to be filled with each iteration of kfold
        for train_index, test_index in split:
```